src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9539,11 +9539,17 @@ private bool ScanDesignator()
 
         private bool IsPossibleLambdaExpression(Precedence precedence)
         {
-            if (precedence <= Precedence.Lambda && this.PeekToken(1).Kind == SyntaxKind.EqualsGreaterThanToken)
+            if (precedence <= Precedence.Lambda)
             {
-                return true;
+                if (this.PeekToken(1).Kind == SyntaxKind.EqualsGreaterThanToken)
+                    return true;
+                else if (this.PeekToken(1).Kind == SyntaxKind.ExclamationToken)
+                {
+                    if ((this.PeekToken(2).Kind == SyntaxKind.EqualsGreaterThanToken) ||
+                            (this.PeekToken(2).Kind == SyntaxKind.CommaToken))
```

Comment:
why the added comma check?  this code needs comments giving examples.  This actually seems wrong to me...

Reply:
why the comma check?

Reply:
I believe this would be for the multiple parameter case with an exclamation:
`Func<int, int, bool> func1 = (x!, y) => x == y;`